src/main/scala/org/qslib/quantscale/time/Frequency.scala

```diff
@@ -47,20 +47,25 @@ package org.qslib.quantscale.time
  * @author Choucri FAHED
  * @since 1.0
  */
-sealed trait Frequency {
-  def apply(): Int
-}
+class Frequency(val value: Int) extends AnyVal
+
+object Frequency {
+
+  def apply(frequency: Int): Frequency = new Frequency(frequency)
 
-case object NoFrequency extends Frequency { override val apply = -1 }
-case object Once extends Frequency { override val apply = 0 }
-case object Annual extends Frequency { override val apply = 1 }
-case object Semiannual extends Frequency { override val apply = 2 }
-case object EveryFourthMonth extends Frequency { override val apply = 3 }
-case object Quarterly extends Frequency { override val apply = 4 }
-case object Bimonthly extends Frequency { override val apply = 6 }
-case object Monthly extends Frequency { override val apply = 12 }
-case object EveryFourthWeek extends Frequency { override val apply = 13 }
-case object Biweekly extends Frequency { override val apply = 26 }
-case object Weekly extends Frequency { override val apply = 52 }
-case object Daily extends Frequency { override val apply = 365 }
-case object OtherFrequency extends Frequency { override val apply = 999 }
+  implicit def unwrapp(frequency: Frequency): Int = frequency.value
```

Comment:
Convenient but also potentially dangerous as you compromise type safety. :)

However, if you want to go all the way you could also have an implicit for `Int => Frequency` (which is even more dangerous).